src/sentry/utils/tag_normalization.py

```diff
@@ -0,0 +1,92 @@
+import functools
+import re
+
+_KNOWN_TAGS = {
+    "sentry.cocoa",
+    "sentry.dart",
+    "sentry.dotnet",
+    "sentry.elixir",
+    "sentry.go",
+    "sentry.java",
+    "sentry.javascript.angular",
+    "sentry.javascript.browser",
+    "sentry.javascript.capacitor",
+    "sentry.javascript.cordova",
+    "sentry.javascript.deno",
+    "sentry.javascript.electron",
+    "sentry.javascript.ember",
+    "sentry.javascript.gatsby",
+    "sentry.javascript.miniapp",
+    "sentry.javascript.nextjs",
+    "sentry.javascript.node",
+    "sentry.javascript.node.experimental",
+    "sentry.javascript.node.hapi",
+    "sentry.javascript.raycast",
+    "sentry.javascript.react",
+    "sentry.javascript.react.native",
+    "sentry.javascript.remix",
+    "sentry.javascript.serverless",
+    "sentry.javascript.sfcc",
+    "sentry.javascript.svelte",
+    "sentry.javascript.sveltekit",
+    "sentry.javascript.vue",
+    "sentry.kubernetes",
+    "sentry.lua",
+    "sentry.native.android",
+    "sentry.native.dotnet",
+    "sentry.native.unity",
+    "sentry.native.unreal",
+    "sentry.objc",
+    "sentry.perl",
+    "sentry.php",
+    "sentry.python",
+    "sentry.ruby",
+    "sentry.rust",
+    "sentry.swift",
+}
+
+
+_SYNONYMOUS_TAGS = {
+    "sentry.android": "sentry.native.android",
```

Comment:
`sentry.native.android` would be one of:

-  `sentry.native.android`
-  `sentry.native.android.capacitor`
-  `sentry.native.android.flutter`
-  `sentry.native.android.react-native`
-  `sentry.native.android.unity`
-  `sentry.native.android.unreal`

Reply:
Yeah, lemme check, but my gut says each of those should be collapsed into its fourth component, not the other way around.

Reply:
Yeah, the only place the capacitor value is set is in the Capacitor SDK, which is separate from our android SDK (which is technically part of our Java SDK, though I do think we should keep it separate).

https://github.com/getsentry/sentry-capacitor/blob/597e9d10cc307bd65957c2754fa0329557b02863/android/src/main/java/io/sentry/capacitor/SentryCapacitor.java#L39-L40

It's a tough call I guess, though. From the perspective of, say, an `event.created` metric, we'd for example probably want to group all the flutter events together, since they're all coming from the same SDK. From a `groups.merged` metric, though, it would probably be more useful to know that native events, across SDKs, are the ones which aren't grouping well and causing people to merge issues. 🤔 

WDYT?

Reply:
Re: Java vs Native, as far as I understand it's whether executed in Java/Kotlin/other-JVM vs Android NDK (C++). 
So if we'd like to lets say group flutter together, `sentry.java` cannot be completely collapsed either. 

Java/Android values
```
 'sentry.java.android',
 'sentry.java.android.capacitor',
 'sentry.java.android.dotnet',
 'sentry.java.android.flutter',
 'sentry.java.android.kmp',
 'sentry.java.android.react-native',
 'sentry.java.android.timber',
 'sentry.java.android.unity',
 'sentry.java.android.unreal',
 ```

Reply:
Right - we have a number of SDKs which have a native component. The RN SDK, for example, will report errors from both its react "frontend" and the native code called by its "backend." 

So if we want to know the answer to questions like, "Which SDK is sending the most events?", then it makes sense to consider anything with `flutter` in the string as part of the Flutter SDK, and look at the `event.created` metric. OTOH, if we want to answer questions like, "Which platforms have bad enough default grouping that people are constantly having to merge issues?," then it makes sense to consider all native events together and look at the `events.merged` metric, since native stacktraces (on which we group) are fundamentally different than JS stacktraces.

I'm realizing in a perfect world we'd do it both ways, tailored to which metric we're adding the info, but at that point we're almost certainly over-engineering it. Which, TBH, we're probably nearing even with this convo. So, all things considered, I think that since the current impetus is to measure grouping effectiveness, you're probably right to group those all under `native`. I think we should add a TODO there, though, along the lines of `Should we be grouping by origin SDK instead? (For example, should we be combining all flutter events rather than all native events?) See XXX`. (I'll insert the link to this comment once I submit it and it gets assigned one.)